microservices/services/members-service/src/main/java/org/xcolab/service/members/domain/member/MemberDaoImpl.java

```diff
@@ -147,6 +154,33 @@ public MemberDaoImpl(DSLContext dslContext) {
         return query.fetchInto(Member.class);
     }
 
+    private SortField[] getDisplayNameSortFields(MemberTable member,
+            boolean isAscendingSortOrder) {
+        // For each record:
+        // If first name != null, sort by trimmed first name (and last name),
+        // else if last name != null, sort by last name
+        // else sort by screen name.
+
+        // nvl(a, b): if (a != null): a, else: b
+        // nullif(a, b): if (a != b): a, else: null
+        // ltrim(a): trim String a from the left
+
+        Field<String> firstName = nullif(ltrim(member.FIRST_NAME), "");
+        Field<String> lastName = nullif(ltrim(member.LAST_NAME), "");
+        Field<String> sortName = nvl(firstName, nvl(lastName, member.SCREEN_NAME));
```

Comment:
Nice use of these cryptic SQL functions! These more complicated sorts in SQL are always a pain.

I find it a bit confusing that the method first creates a sort name, which is either the first, last, or screen name, and then sorts by sortName and lastName. What do you think about building a single "displayName" field similarly to the way it's built in `Member#getDisplayName()`? Simplified Java code from that method:
```java
String fullName = this.getFirstName()+ " " + this.getLastName();
return isNotBlank(fullName) ? fullName : screenName;
```

It's likely a bit uglier than that in SQL, but I think the purpose will be clearer. You can use the coalesce function to turn nulls into empty strings before concatenating firstName and lastName and then use a similar mechanism as above to use screenName if fullName is empty.

I think this would make it easier to understand the sort logic.

Reply:
Done. It should be more legible now.